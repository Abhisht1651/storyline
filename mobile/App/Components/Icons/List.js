import React from 'react'
import { View, Text, TouchableOpacity } from 'react-native'

// Styles
import styles from '../Styles/IconStyles/ListStyles'
import RavenSVG from '../../Images/Icons/Path-1.svg'
import ShareSVG from '../../Images/Icons/Path-3.svg'
import ReactSVG from '../../Images/Icons/Path.svg'

const List = () => {
  return (
    <View style={styles.cardBox}>
      <TouchableOpacity style={styles.rowBox}>
        <View style={styles.profileIconBox}>
          <ShareSVG />
        </View>
        <View style={styles.TextBox}>
          <Text style={styles.TextStyles}>Share</Text>
        </View>
      </TouchableOpacity>

      <TouchableOpacity style={styles.rowBox2}>
        <View style={styles.dimen}>
          <RavenSVG />
        </View>
        <View style={styles.TextBox}>
          <Text style={styles.TextStyles}>Raven</Text>
        </View>
      </TouchableOpacity>

      <TouchableOpacity style={styles.rowBox1}>
        <View style={styles.profileIconBox}>
          <ReactSVG />
        </View>
        <View style={styles.TextBox}>
          <Text style={styles.TextStyles}>React</Text>
        </View>
      </TouchableOpacity>
    </View>
  )
}

export default List
