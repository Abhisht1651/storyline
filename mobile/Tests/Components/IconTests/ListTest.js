import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'

import List from '../../../App/Components/Icons/List'

test('List Icon component renders correctly', () => {
  const tree = renderer.create(<List />).toJSON()
  expect(tree).toMatchSnapshot()
})
