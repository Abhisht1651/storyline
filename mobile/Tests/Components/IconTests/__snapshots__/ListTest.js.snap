// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Icon component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "flexDirection": "row",
      "height": 18,
      "opacity": 1,
      "width": 274,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": 16,
          "width": 19,
        }
      }
    >
      <SvgMock />
    </View>
    <View
      style={
        Object {
          "height": 18,
          "width": 46,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 16,
            "letterSpacing": -0.09,
            "opacity": 1,
          }
        }
      >
        Share
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 44,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": 16,
          "width": 17,
        }
      }
    >
      <SvgMock />
    </View>
    <View
      style={
        Object {
          "height": 18,
          "width": 46,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 16,
            "letterSpacing": -0.09,
            "opacity": 1,
          }
        }
      >
        Raven
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 37,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "height": 16,
          "width": 19,
        }
      }
    >
      <SvgMock />
    </View>
    <View
      style={
        Object {
          "height": 18,
          "width": 46,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#1F1F60",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 16,
            "letterSpacing": -0.09,
            "opacity": 1,
          }
        }
      >
        React
      </Text>
    </View>
  </View>
</View>
`;
